packages/components/src/avatar/avatar.tsx

```diff
@@ -1,97 +1,356 @@
+                  >
+                    {props.name.slice(0, 2).toUpperCase()}
+                  </Text>
+                )}
+              </Fallback>
+            )}
+          </>
+        )}
+
+        {props.type === 'wallet' && (
+          <Fallback borderRadius={radius} backgroundColor={backgroundColor}>
+            <Text size={textSizes[size]} weight="medium" color="$white-100">
+              {props.name.slice(0, 2).toUpperCase()}
+            </Text>
+          </Fallback>
+        )}
+
+        {props.type === 'channel' && (
+          // fixme: Type 'undefined' is not assignable to type '32 | 11 | 13 | 15 | 19 | 27'
+          <Text size={emojiSizes[size] as TextProps['size']}>
+            {props.emoji}
+          </Text>
+        )}
+
+        {props.type === 'icon' && (
+          <>
+            {cloneElement(props.icon, { color: props.color ?? '$white-100' })}
+          </>
+        )}
+      </Base>
+
+      {props.type === 'user' && props.indicator && (
         <Unspaced>
-          <Indicator size={size} state={indicator} />
+          <Indicator size={size} state={props.indicator} />
         </Unspaced>
       )}
-      <Shape shape={shape}>
-        <Image
-          src={src}
-          width="full"
-          aspectRatio={1}
-          onLoad={() => setStatus('loaded')}
-          onError={() => setStatus('error')}
-        />
-
-        {status === 'error' && (
-          <Fallback
-            width={size}
-            height={size}
-            display="flex"
-            alignItems="center"
-            justifyContent="center"
-          >
-            PP
-          </Fallback>
-        )}
-      </Shape>
-    </Base>
+
+      {props.type === 'channel' && hasLock(props) && (
+        <LockBase variant={props.size}>
+          {props.lock === 'locked' ? <LockedIcon /> : <UnlockedIcon />}
+        </LockBase>
+      )}
+    </Stack>
   )
 }
 
 export { Avatar }
-export type { Props as AvatarProps }
+export type { AvatarProps }
 
 const Base = styled(Stack, {
   name: 'Avatar',
 
   position: 'relative',
   justifyContent: 'center',
   alignItems: 'center',
+  overflow: 'hidden',
 
   variants: {
-    // defined in Avatar props
     size: {
-      '...': (size: number) => {
-        return {
-          width: size,
-          height: size,
-        }
+      80: {
+        width: 80,
+        height: 80,
       },
-    },
-
-    shape: {
-      circle: {
-        borderRadius: '$full',
+      56: {
+        width: 56,
+        height: 56,
+      },
+      48: {
+        width: 48,
+        height: 48,
       },
-      rounded: {
-        borderRadius: '$16',
+      32: {
+        width: 32,
+        height: 32,
+      },
+      28: {
+        width: 28,
+        height: 28,
+      },
+      24: {
+        width: 24,
+        height: 24,
+      },
+      20: {
+        width: 20,
+        height: 20,
+      },
+      16: {
+        width: 16,
+        height: 16,
       },
     },
 
```

Comment:
I know outline was here before, but I am not sure how I feel about this since it does not exist in design system. I am thinking that it should be added in the sidebar by parent, because it's the only place where it is used like that. I will let you and @jkbktl decide.

Reply:
I'm okay to remove outline, but I'd probably ask designers to make sure they don't want to use it elsewhere, but sidebar.

Reply:
Raised at https://discord.com/channels/@me/1079757666747940965/1097455683600928822.